### DISCUSSION TITLE
I want to use sklearn module.

### DISCUSSION THREAD

**Discussion by kindamu24005** [getredash/redash#6635] [Q&A] [answered]
Thank you for checking my question.

I want to use sklearn module.
<img width="1136" alt="スクリーンショット 2023-11-30 11 06 01" src="https://github.com/getredash/redash/assets/107189394/0010e06f-51cb-4fae-b8d5-6e9da18dca9a">

How should I fill in "Modules to import prior to running the script"?
<img width="388" alt="スクリーンショット 2023-11-30 11 08 06" src="https://github.com/getredash/redash/assets/107189394/3cc292aa-b9cd-4a29-931c-18e1da900103">

**Comment by guidopetri** [comment]
`sklearn` is [not installed because we don't use it](https://github.com/getredash/redash/blob/master/pyproject.toml), so you are not going to be able to run sklearn in redash.

**Reply by kindamu24005** [reply]
Thank you for your answer.

If I add scikit-learn to pyproject.toml and do "docker-compose up -d", can I use it?

**Reply by guidopetri** [reply, marked as answer]
You would have to rebuild the image, which is not super easy, but should be doable. If you want to do that, you should look at the CI instructions for how to build it, and you would have to maintain your own fork of redash.

I would not recommend this, since redash isn't really meant for running machine learning models within it. The docker containers will probably be very slow, anyway.
